Signal/src/call/PeerConnectionClient.swift

```diff
@@ -326,115 +404,145 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
 
     // MARK: - Data Channel
 
-    func sendDataChannelMessage(data: Data) -> Bool {
-        guard let dataChannel = self.dataChannel else {
-            Logger.error("\(TAG) in \(#function) ignoring sending \(data) for nil dataChannel")
-            return false
+    public func sendDataChannelMessage(data: Data) -> Promise<Bool> {
```

Comment:
The only API change of significance is that sendDataChannelMessage() now returns a promise.